types/artifacts.d.ts

```diff
@@ -743,6 +741,56 @@ declare global {
         /** Column number in the script (0-based). */
         columnNumber: number;
       }
+
+      /** Describes a generic console message. */
+      interface BaseConsoleMessage {
+        /**
+         * What emitted the console message (console API, violation log, etc.).
+         * See subtypes for possible values.
+         */
+        source: string,
+        /** The log level: info, warning, etc. See subtypes for possible values. */
+        level: string,
+        /**
+         * The text printed to the console, as shown on the browser. Note objects printed to the
+         * console will be stringified if primitive or JSON-serializable. Otherwise it will be
+         * a String representation of the object.
+         */
+        text: string;
+        /** Time of the console log in milliseconds since epoch. */
+        timestamp: number;
+        /** The stack trace of the log/exception, if known. */
+        stackTrace?: Crdp.Runtime.StackTrace;
+        /** The URL of the log/exception, if known. */
+        url?: string;
+        /** Line number in the script (0-based), if known. */
+        lineNumber?: number;
+        /** Column number in the script (0-based), if known. */
+        columnNumber?: number;
+      }
+
+      /** Describes a console message logged by a script using the console API. */
+      interface ConsoleAPICall extends BaseConsoleMessage {
+        /** The console API invoked. Only the following console API calls are gathered. */
+        source: 'console.warn' | 'console.error';
```

Comment:
I like this explicitness 👍

Reply:
I'm walking this back a smidge after seeing the impl that we don't actually get the API that was called and we're inferring this from the level, see https://github.com/GoogleChrome/lighthouse/pull/11663#discussion_r525578538 for more